app/lib/shared/integrity.dart

```diff
@@ -35,8 +36,18 @@ class IntegrityChecker {
   IntegrityChecker(this._db, {int? concurrency})
       : _concurrency = concurrency ?? 1;
 
-  /// Runs integrity checks, and reports the list of problems.
-  Stream<String> check() async* {
+  /// Runs integrity checks, and reports the problems via a [Logger].
+  Future<void> verifyAndLogIssues() async {
+    var count = 0;
+    await for (final problem in findProblems()) {
+      count++;
+      _logger.reportError(problem);
```

Comment:
Should we prefix these to make them easier to filter for?

Reply:
Wouldn't the logger's name enough for that?